### PR TITLE
feat(mobile-exp): Add new issue details tag summary for mobile platforms 

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Adds a new wip Tag Summary for mobile platform issues to the top of the issue details sidebar.
This new ui is behind the `issue-details-tag-improvements` feature flag and only appears if the issue project is a mobile platform.
image

### CODE REVIEW

File: static/app/components/group/tagFacets.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                      setSelectedTag(tagKey);
+                    }}
+                  >
+                    {tagKey}
+                  </Button>
+                );
+              })}
+            </ButtonBar>
+            <BreakdownBarsContainer>
+              <BreakdownBars data={points ?? []} />
```
Comment (narsaynorath) [inline static/app/components/group/tagFacets.tsx:80]: Could always just add `?? []` where `points` is defined so it's always an array and we don't forget to add `?? []` if its used again

File: static/app/components/group/tagFacets.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      setLoading(false);
+    });
+  }, [api, environments, groupId, tagKeys]);
+
+  const availableTagKeys = tagKeys.filter(tagKey => tagsData[tagKey]!!);
+  const points = tagsData[selectedTag]?.topValues.map(({name, count}) => {
+    return {label: name, value: count};
+  });
+
+  if (!loading) {
```
Comment (narsaynorath) [inline static/app/components/group/tagFacets.tsx:57]: Small nit-pick, but I find it nice to get out of the function quick and do:
```js
if (loading) {
  return <Placeholder ... />;
}

if (availableTagKeys.length === 0) {
  return null;
}

// the rest
```
to reduce some nesting and separate out the valid states from anything "invalid"

**Review by narsaynorath — APPROVED** [review]
Some small comments but looks good to me! Tested it out locally and it's nice :fire:

File: static/app/components/group/tagFacets.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    render(<TagFacets environments={[]} groupId="1" tagKeys={MOBILE_TAGS} />);
+    await waitFor(() => {
+      expect(tagsMock).toHaveBeenCalled();
+    });
+    expect(screen.queryByText('os')).not.toBeInTheDocument();
+    expect(screen.queryByText('device')).not.toBeInTheDocument();
+    expect(screen.queryByText('release')).not.toBeInTheDocument();
+  });
+
+  it('displays os tags', async function () {
```
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply static/app/components/group/tagFacets.spec.tsx]: 👍 merged the tests!

**Pull request merged**